server/src/main/java/com/paypal/selion/pojos/BrowserInformationCache.java

```diff
@@ -0,0 +1,242 @@
+|  Copyright (C) 2016 PayPal                                                                                          |
+|                                                                                                                     |
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.pojos;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+
+import org.openqa.grid.internal.Registry;
+import org.openqa.grid.internal.RemoteProxy;
+
+import com.paypal.selion.logging.SeLionGridLogger;
+
+/**
+ * <code>BrowserInformationCache</code> acts as a cache holding the browser name and the maximum instances available in
+ * a particular node. The key to the cache is the {@link URL} object of the node that uniquely identifies the node.
+ */
+public class BrowserInformationCache {
+
+    // List of supported browsers from
+    // https://code.google.com/p/selenium/wiki/DesiredCapabilities
+    public static final String[] SUPPORTED_BROWSERS = { "android", "chrome", "firefox", "htmlunit",
+            "internet explorer", "iPhone", "iPad", "opera", "safari" };
+
+    private static final int INITIAL_CAPACITY = 25;
+
+    private static final SeLionGridLogger logger = SeLionGridLogger.getLogger(BrowserInformationCache.class);
+
+    private static final BrowserInformationCache instance = new BrowserInformationCache();
+
+    private Map<URL, TestSlotInformation> nodeMap = null;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid using redundant field initializer for 'nodeMap'](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103178711/issues/source?bid=2928425&fileBranchId=3143459#l50)